yoyodyne/models/lstm.py

```diff
@@ -149,6 +149,8 @@ def decode(
         predictions = torch.stack(predictions)
         return predictions
 
+    # TODO(adamits,kbg): out-of-date documentation.
```

Comment:
Is this for the README or the method below? Or perhaps just a general note for me to check all of it out?

Reply:
I would like help explaining the return type, which is currently just labeled "_description_". I *think* the union is because you can request confidences in addition to just the prediction tensor but I'm not sure. Could you confirm and then I can 1) fill this in and 2) remove the TODO?

I also want to get @bonham79's suggestion on that one undocumented static function into this PR.

Reply:
> I would like help from you in explaining the return type, which is currently just labeled "_description_". I _think_ the union is because you can request confidences in addition to just the prediction tensor but I'm not sure. Could you confirm and then I can 1) fill this in and 2) remove the TODO?
> 
> I also want to get @bonham79's suggestion on that one undocumented static function into this PR.

Ah I see. I think it should be a List of ints (the predictions) and a optionally a List of floats (the likelihoods). I can rewrite this so its easier to follow, but I think I have a long-standing TODO to implement this to handle batches for a bunch of the architectures. Right the Union was because you can optionally `return_confidences`, but I don't think thats the right usage of that type hint given we can just mark an optional return.

I also think we should just remove `return_confidences` and always return them. It is easy to `preds, _ = model.beam_decode(...)`

Reply:
Okay if I just add a TODO to the effect "Modify to eliminate polymorphic return type; always return confidences." ?

Reply:
> Okay if I just add a TODO to the effect "Modify to eliminate polymorphic return type; always return confidences." ?

Yeah sure